yt/visualization/volume_rendering/scene.py

```diff
@@ -287,7 +302,11 @@ def save(self, fname=None, sigma_clip=None):
             suffix = '.png'
             fname = '%s%s' % (fname, suffix)
 
-        self.render()
+        # in most cases we want to render every time, but in some cases pulling
+        # the previous render is desirable (e.g., if only changing sigma_clip or
+        # saving after a call to sc.show()).
+        if render or hasattr(self,'_last_render') is False:
```

Comment:
For point 1 - Great suggestion! Yes, adding `_last_render` as an attribute at init will make it clearer. 

For point 2 - I like adding the message. Since whether or not to call `render()` before `save()` may be confusing for new users (hopefully the updated docs help!), would it be helpful to add a second message to notify that the most recently rendered image is being used? i.e., 

```
if not render and self._last_render is None:
    mylog.warning("No previous rendered image found, rendering now.")
    render = True
else: 
    mylog.warning("Saving most recently rendered image") 
```

Or is that too much info?

Reply:
Not too much info imo, I like it, although it might just be an "info" level logging message, instead of a warning :)

Reply:
Since adding the `if` statement adds a few lines and those lines need to also happen in `save_annotated`, any objection to creating a new function to avoid copy/pasting the 6-7 lines, e.g., `_check_render()`? 

I'm also thinking that it may be better to further split up the logging into: 

```
        if self._last_render is None:
            mylog.warning("No previous rendered image found, rendering now.")
            render = True
        elif render:
            mylog.info("Overwriting previous rendered image with new rendering.")
        else:            
            mylog.info("Saving most recently rendered image.")
```

Maybe the second condition should be a `warning` as well? Since it is a desired behavior, `info` feels appropriate but happy to change it.

Reply:
No objection, on the contrary. If you need to reuse code it's _much_ better to make it... reusable. :-)
I think your idea to further split this if statement is great. I may have a small preference for a warning in the `elif render` case but your choice !